File: hackathon/security-copilot/backend/app/schemas/approval.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ApprovalBase(BaseModel):
    action: str
    notes: Optional[str] = None


class ApprovalCreate(BaseModel):
    notes: Optional[str] = None


class ApprovalResponse(BaseModel):
    id: int
    article_id: int
    analyst_id: Optional[int] = None
    action: str
    notes: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ApprovalListResponse(BaseModel):
    items: list[ApprovalResponse]
    total: int

File: hackathon/security-copilot/frontend/src/types/approval.ts
export interface Approval {
  id: number;
  article_id: number;
  analyst_id: number | null;
  action: 'approved' | 'rejected' | 'under_review';
  notes: string | null;
  created_at: string;
}

export interface ApprovalListResponse {
  items: Approval[];
  total: number;
}

export interface ApprovalListParams {
  article_id?: number;
  skip?: number;
  limit?: number;
}
